packages/ccl-docs/src/content/docs/behavior-reference.md

```diff
@@ -0,0 +1,299 @@
+
+## Behavior Groups
+
+### Line Endings
+
+**Options:** `crlf_preserve_literal` vs `crlf_normalize_to_lf`
+
+Controls how Windows-style line endings (CRLF, `\r\n`) are handled during parsing.
+
+#### `crlf_normalize_to_lf`
+
+Converts all CRLF sequences to LF (`\n`) before or during parsing. This is the most common choice for cross-platform compatibility.
+
+```ccl
+key = value\r\n
+nested = \r\n
+  child = data\r\n
+```
+
+**Result:** The `\r` characters are stripped, values contain only `\n`.
+
+```json
+{"key": "value", "nested": {"child": "data"}}
+```
+
+#### `crlf_preserve_literal`
+
+Preserves `\r` characters exactly as they appear in the input. Values may contain literal carriage return characters.
+
+```ccl
+key = value\r\n
+```
+
+**Result:** The value includes the `\r` character.
+
+```json
+{"key": "value\r"}
+```
+
+**Recommendation:** Most implementations should use `crlf_normalize_to_lf` for predictable cross-platform behavior.
+
+---
+
+### Boolean Parsing
+
+**Options:** `boolean_lenient` vs `boolean_strict`
+
+Controls which string values are accepted as booleans by typed access functions like `get_bool`.
+
+#### `boolean_strict`
+
+Only accepts `true` and `false` (case-sensitive) as boolean values.
+
+```ccl
+enabled = true
+disabled = false
+active = yes
+```
+
+```javascript
+getBool(obj, "enabled")   // → true
+getBool(obj, "disabled")  // → false
+getBool(obj, "active")    // → ERROR: "yes" is not a valid boolean
+```
+
+#### `boolean_lenient`
+
+Accepts additional truthy/falsy values beyond `true`/`false`.
+
+| Truthy Values | Falsy Values |
+|---------------|--------------|
+| `true`, `yes`, `on`, `1` | `false`, `no`, `off`, `0` |
+
+```ccl
+enabled = yes
+disabled = no
+active = on
+inactive = off
+```
+
+```javascript
+getBool(obj, "enabled")   // → true
+getBool(obj, "disabled")  // → false
+getBool(obj, "active")    // → true
+getBool(obj, "inactive")  // → false
+```
+
+**Note:** Both modes accept `true` and `false`. The difference is whether additional values like `yes`/`no` are also accepted.
+
+**Recommendation:** Use `boolean_strict` for stricter validation, `boolean_lenient` for more user-friendly configuration files.
+
+---
+
+### Tab Handling
+
+**Options:** `tabs_preserve` vs `tabs_to_spaces`
+
+Controls how tab characters are processed during parsing.
+
+#### `tabs_preserve`
+
+Tab characters are kept as-is in values and used literally for indentation calculation.
+
+```ccl
+key = \tindented value
+nested =
+\tchild = data
+```
+
+**Result:** Tab characters remain in the parsed output.
+
+#### `tabs_to_spaces`
+
+Tab characters are converted to spaces (typically 4 spaces per tab) before parsing.
+
+```ccl
+key = \tindented value
+```
+
+**Result:** `key` has value `    indented value` (tab converted to spaces).
+
+**Recommendation:** `tabs_to_spaces` provides more predictable behavior across editors with different tab width settings.
```

Comment:
Not accurate. tabs_to_sapces means treating a tab character as a SINGLE space - basically `\t` and ` ` are the same width for indentation purposes.